benchmarks/miscellaneous/nondet_aligned_alloc.c

```diff
@@ -0,0 +1,24 @@
+#include <stdint.h>
+#include <assert.h>
+#include <dat3m.h>
+#include <stdlib.h>
+
+int main()
+{
+    int size_int = __VERIFIER_nondet_int();
+    int align_int = __VERIFIER_nondet_int();
+    __VERIFIER_assume(size_int > 0);
+    __VERIFIER_assume(align_int > 0);
+    size_t size = (size_t)size_int;
+    size_t align = (size_t)align_int;
+    int *array = malloc(42 * sizeof(int));
+    int *array_2 = aligned_alloc(align, size * sizeof(int));
+    // NOTE: making the first allocation non-det-sized makes the solver extremely slow to verify the
+    // custom aligned allocation. Even when fixing the alignment to a constant it gets super slow if that constant
+    // is not a power of two. In other words, aligning variable-sized addresses is very expensive currently, especially
+    // for non-power-of-two alignments (those don't exist in practice though).
+
+    assert (array != array_2);
```

Comment:
You want to assert a certain distance between the arrays? I can add that. Overflows do not matter in this example because I generate positive 32-bit sizes but we have a 64-bit address space so even MAX_INT does not overflow.
But this conversion is sublte because the compiler sign-extends but I want a zero-extension, which I get by using positive values (for those the sign bit is 0 so sign and zero-extension do the same).

Reply:
I want to assert that the arrays do not overlap.

Reply:
Done. The assertion assumes a certain ordering between the addresses though. I didn't bother looking for the abs function and somehow I have the feeling that it would involve unsupported intrinsics :P